knowledge-base/scheduler-conditionally-disable-drag-for-some-appointments.md

```diff
@@ -0,0 +1,151 @@
+
+    private DateTime DayStart { get; set; } = new DateTime(2000, 1, 1, 8, 0, 0);
+
+    private void OnItemRenderHandler(SchedulerItemRenderEventArgs e)
+    {
+        SchedulerAppointment appt = e.Item as SchedulerAppointment;
+        
+        //add custom class to stop the pointer events of the appointment
+        if (appt.NonDraggable)
+        {
+            e.Class = "non-draggable-appointment";
+        }
+    }
+
+    private void OnUpdate(SchedulerUpdateEventArgs args)
+    {
+        SchedulerAppointment item = (SchedulerAppointment)args.Item;
+
+        if (item.NonDraggable)
+        {
+            //Do not execute code for updating the appointment.
+        }
+        else
+        {
+            //Execute appointment update.
+            //The actual update is not handled here for brevity.
+        }
+    }
+
+    private List<SchedulerAppointment> Appointments = new List<SchedulerAppointment>()
+    {
+            new SchedulerAppointment
+            {
+                Title = "Vet visit",
+                Description = "The cat needs vaccinations and her teeth checked.",
+                Start = new DateTime(2019, 11, 26, 11, 30, 0),
+                End = new DateTime(2019, 11, 26, 12, 0, 0),
+                NonDraggable = true
+            },
+
+            new SchedulerAppointment
+            {
+                Title = "Planning meeting",
+                Description = "Kick off the new project.",
+                Start = new DateTime(2019, 11, 25, 9, 30, 0),
+                End = new DateTime(2019, 11, 25, 12, 45, 0)
+            },
+
+            new SchedulerAppointment
+            {
+                Title = "Trip to Hawaii",
+                Description = "An unforgettable holiday!",
+                IsAllDay = true,
+                Start = new DateTime(2019, 11, 29),
+                End = new DateTime(2019, 12, 07),
+                NonDraggable = true
+            },
+
+            new SchedulerAppointment
+            {
+                Title = "Online conference",
+                IsAllDay = true,
+                Start = new DateTime(2019, 11, 25),
+                End = new DateTime(2019, 11, 26)
+            }
+    };
+
+    public class SchedulerAppointment
+    {
+        public string Title { get; set; }
+        public string Description { get; set; }
+        public DateTime Start { get; set; }
+        public DateTime End { get; set; }
+        public bool IsAllDay { get; set; }
+
+        public bool NonDraggable { get; set; }
+    }
+}
+
+<style>
+    .non-draggable-appointment {
+        background-color: grey;
+        font-weight: 900;
+        pointer-events: none !important;
+    }
+</style>
+````
+
+## Notes
+
+- The custom CSS class used to disable dragging should be defined in your application's stylesheets.
+- Ensure the condition for disabling drag is consistent across both the `OnItemRender` event and the `OnUpdate` event handler.
```

Comment:
Consider adding a See Also section with relevant links.

Reply:
The general practice for the KB articles is to link the relevant resources in the text instead of having an additional See Also section. The majority of the KB articles do not have that. Maybe we can additionally synchronize whether we want to adopt such section in the KB structure.